cmd/juju/application/bundle/bundle.go

```diff
@@ -345,3 +350,55 @@ func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints fu
 	}
 	return errors.Trace(verifyError)
 }
+
+func verifyMixedSeriesBasesMatch(data *charm.BundleData) error {
+	if data == nil {
+		return nil
+	}
+	if data.Series != "" && data.DefaultBase != "" {
+		b, err := corebase.ParseBaseFromString(data.DefaultBase)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		s, err := corebase.GetSeriesFromBase(b)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if s != data.Series {
+			return errors.NewNotValid(nil, fmt.Sprintf("bundle series %q and base %q must match if supplied", data.Series, data.DefaultBase))
+		}
+	}
+
+	for name, m := range data.Machines {
+		if m != nil && m.Series != "" && m.Base != "" {
+			b, err := corebase.ParseBaseFromString(m.Base)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			s, err := corebase.GetSeriesFromBase(b)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if s != m.Series {
+				return errors.NewNotValid(nil, fmt.Sprintf("machine %q series %q and base %q must match if supplied", name, m.Series, m.Base))
```

Comment:
s/must match if supplied/must match if both supplied/

In all three errors please.

Reply:
done